src/js/vic-v2/components/PhotoField.jsx

```diff
@@ -284,40 +310,99 @@ export default class PhotoField extends React.Component {
+    if (zoomValue < this.state.minRatio) {
+      zoomToBoundaryRatio = this.state.minRatio;
+    } else if (zoomValue > this.state.maxRatio) {
+      zoomToBoundaryRatio = this.state.maxRatio;
+    }
 
-      this.setState({
-        warningMessage: makeCropBoundaryWarningMessage({ zoomWarn, ...moveBoundariesMet }),
-        ...makeMoveButtonsEnabledStates(moveBoundariesMet)
-      });
-    } else {
-      e.preventDefault();
+    // force zoom within zoom bounds
+    if (zoomToBoundaryRatio) {
+      this.refs.cropper.zoomTo(zoomToBoundaryRatio); // force zoom within constraints
+      e.preventDefault(); // prevents bad zoom attempt
+      return; // don't update state until the subsequent zoom attempt
     }
+
+    // zoom value is good- update the state / messaging
+    this.setState({ zoomValue });
+
+    const zoomWarn = zoomValue >= WARN_RATIO;
+    // at this point, the onZoom event has not resized the canvas-
+    // this forces the canvas back into move bounds if the zoom pulls a canvas edge into the cropBox
+    //   after the canvas has rendered with new width values
+    window.requestAnimationFrame(() => this.maybeMoveCanvasWithinBounds({}, zoomWarn));
   }
 
   setCropBox = () => {
-    const cropper = this.refs.cropper;
-    const slider = this.refs.slider;
-    const { width, naturalWidth } = this.refs.cropper.getImageData();
-    slider.value = width / naturalWidth;
-    const containerData = cropper.getContainerData();
-    const containerWidth = containerData.width;
-    const smallScreen = isSmallScreen(this.state.windowWidth);
-    const cropBoxSize = smallScreen ? 240 : 300;
-    const cropBoxLeftOffset = (containerWidth - cropBoxSize) / 2;
-    const cropBoxData = {
-      left: cropBoxLeftOffset,
-      top: 0,
-      width: cropBoxSize,
-      height: cropBoxSize
-    };
-    cropper.setCropBoxData(cropBoxData);
+    window.requestAnimationFrame(() => {
+      // center the cropbox using container width
+      const containerWidth = this.refs.cropper.getContainerData().width;
+      // const containerWidth = this.refs.container.offsetWidth;
+      const smallScreen = isSmallScreen(this.state.windowWidth);
+      const left = smallScreen ? (containerWidth / 2) - 120 : (containerWidth / 2) - 150;
+      this.refs.cropper.setCropBoxData({
+        top: 0,
+        left,
+        height: smallScreen ? 240 : 300,
+        width: smallScreen ? 240 : 300
+      });
+
+      // use the cropbox dimensions to force canvas into default position
+      const { height: cropBoxHeight, width: cropBoxWidth, left: cropBoxLeft } = this.refs.cropper.getCropBoxData();
+      const { width: oldCanvasWidth, height: oldCanvasHeight, naturalHeight, naturalWidth } = this.refs.cropper.getCanvasData();
+      // tall images take the full width of the cropBox
+      if (naturalHeight > naturalWidth) {
+        this.refs.cropper.setCanvasData({
+          width: cropBoxWidth,
+          left: cropBoxLeft,
+          top: 0
```

Comment:
@jbalboni This resets the top coordinate on narrow images after the window is resized. The canvas will now reset to its default position when the window is resized. This should fix the positioning issue you saw. 

Also noticed that error warning messages were not being reset when the window was resized- added that as well.